Source/msg.cpp

```diff
@@ -49,7 +49,7 @@ void msg_send_packet(int pnum, const void *packet, DWORD dwSize)
 	sgpCurrPkt->dwSpaceLeft -= dwSize;
 }
 
-TMegaPkt *msg_get_next_packet()
+void msg_get_next_packet()

```

Comment:
should it be `void *`? Or is this just meant to be overwriting the global?

Reply:
Returning is pointless here, since `p` ends up pointing to `sgpMegaPkt` which is global. None of the calls use the return value. The best way to figure stuff like this out is look at the beta. `eax` is handled differently depending on whether the function is expected to return a value. At the end of each function (when compiled unoptimized) exists a plain `jmp $-5`. This jump separates early returns from reaching the end of the function flow. If a value is to be returned, `eax` must be set directly before any jumps to the end. If no early returns exist, `eax` is stored *after* the return.
![asm](https://user-images.githubusercontent.com/15209402/74078268-ca664c00-49ed-11ea-8391-f09f60937036.PNG)

Reply:
Thanks! I love insights into low-level tidbits like these :)

Reply:
me too :p